tests/publiccloud/boottime.pm

```diff
@@ -0,0 +1,202 @@
+    die("Unable to get system-analyze in $args{timeout} seconds") unless (time() - $start_time < $args{timeout});
+    return extract_startup_timings($output);
+}
+
+=head2 build_influx_kv
+Build a influx-db key value pair with all needed escapings see doc:
+    https://docs.influxdata.com/influxdb/v1.7/write_protocols/line_protocol_tutorial/#special-characters-and-keywords
+=cut
+sub build_influx_kv {
+    my $hash = shift;
+    my $req  = '';
+    for my $k (keys(%{$hash})) {
+        my $v = $hash->{$k};
+        $v =~ s/,/\\,/g;
+        $v =~ s/ /\\ /g;
+        $v =~ s/=/\\=/g;
+        $req .= $k . '=' . $v . ',';
+    }
+    return substr($req, 0, -1);
+}
+
+sub build_influx_query {
+    my $data = shift;
+    my $req  = $data->{table} . ',';
+    $req .= build_influx_kv($data->{tags});
+    $req .= ' ';
+    $req .= build_influx_kv($data->{values});
+    return $req;
+}
+
+sub run {
+    my ($self) = @_;
+
+    $self->select_serial_terminal;
+
+    # $tags and $startup_timings are the key/values which get stored in influxdb.
+    my $tags = {
+        instance_type     => get_required_var('PUBLIC_CLOUD_INSTANCE_TYPE'),
+        os_flavor         => get_required_var('FLAVOR'),
+        os_version        => get_required_var('VERSION'),
+        os_build          => get_required_var('BUILD'),
+        os_pc_build       => get_required_var('PUBLIC_CLOUD_BUILD'),
+        os_pc_kiwi_build  => get_required_var('PUBLIC_CLOUD_BUILD_KIWI'),
+        os_kernel_release => undef,
+        os_kernel_version => undef,
+    };
+    my $startup_timings = {};
+
+    # Is used to specify thresholds per measurement. If one value is exceeded
+    # the test is set to fail.
+    my $thresholds = {
+        # First boot after provisioning
+        kernel     => 15,
+        userspace  => 60,
+        initrd     => 10,
+        overall    => 120,
+        ssh_access => 60,
+
+        # Values after soft reboot
+        kernel_soft     => 15,
+        userspace_soft  => 60,
+        initrd_soft     => 10,
+        overall_soft    => 120,
+        ssh_access_soft => 70,
+
+        # Values after hard reboot
+        kernel_hard     => 15,
+        userspace_hard  => 60,
+        initrd_hard     => 10,
+        overall_hard    => 120,
+        ssh_access_hard => 60,
+    };
+
+    my $provider = $self->provider_factory();
+
+    # Provision the instance
+    my $instance = $provider->create_instance(check_connectivity => 0);
+    my $ssh_time = $instance->check_ssh_port(timeout => 300);
+    $startup_timings->{ssh_access} = $ssh_time;
+
+    my $systemd_analyze = do_systemd_analyze($instance);
+    $startup_timings->{$_} = $systemd_analyze->{$_} foreach (keys(%{$systemd_analyze}));
+
+    # Collect kernel version
+    $tags->{os_kernel_release} = $instance->run_ssh_command(cmd => 'uname -r');
+    $tags->{os_kernel_version} = $instance->run_ssh_command(cmd => 'uname -v');
+
+    # Do soft reboot
+    my ($shutdown_time, $startup_time) = $instance->softreboot();
```

Comment:
you never using `$shutdown_time` - why you returning it ?

Reply:
To make the function convenient.

Reply:
If you don't have hard feelings. I would like to keep this behavior.

Reply:
I do have feelings but they are not so hard . after all I can force you to use "my style" always :) . Let's leave it if you think it is more convenient

Reply:
correction - **CAN NOT** force you :D

Reply:
:D, was it a "freudscher versprecher"?